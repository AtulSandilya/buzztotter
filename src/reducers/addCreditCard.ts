export interface CardDataForVerification {
  cardNumber: string;
  cardExpMonth: string;
  cardExpYear: string;
  cardCvc: string;
}

interface AddCreditCardState {
  attemptingVerification: boolean;
  isVerified: boolean;
  failed: boolean;
  failMessage: string;
}

const initialAddCreditCardState: AddCreditCardState = {
  attemptingVerification: false,
  failMessage: "",
  failed: false,
  isVerified: false,
};

export const addCreditCard = (state = initialAddCreditCardState, action) => {
  switch (action.type) {
    case "ATTEMPTING_CREDIT_CARD_VERIFICATION":
      return {
        ...state,
        attemptingVerification: true,
      };
    case "SUCCESSFUL_CREDIT_CARD_VERIFICATION":
      return {
        ...state,
        attemptingVerification: false,
        isVerified: true,
      };
    case "FAILED_STRIPE_ADD_CARD_TO_CUSTOMER":
    case "FAILED_CREDIT_CARD_VERIFICATION":
      return {
        ...state,
        attemptingVerification: false,
        failMessage: action.payload.error,
        failed: true,
      };
    case "THROWN_STRIPE_ADD_CARD_TO_CUSTOMER":
    case "THROWN_CREDIT_CARD_VERIFICATION":
      return {
        ...state,
        attemptingVerification: false,
        failMessage: state.failMessage.length > 0
          ? state.failMessage
          : action.payload.error,
        failed: true,
      };
    case "END_CREDIT_CARD_VERIFICATION_IF_NOT_ATTEMPTING":
      if (state.attemptingVerification) {
        return state;
      } else {
        return initialAddCreditCardState;
      }
    case "END_CREDIT_CARD_VERIFICATION":
      return initialAddCreditCardState;
    default:
      return state;
  }
};
