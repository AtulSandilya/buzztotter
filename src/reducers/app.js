const initialAppState = {
  isLoading: true,
}

export const app = (state = initialAppState, action) => {
  switch(action.type){
    case 'LOADING_COMPLETE':
      return {...state,
        isLoading: false,
      }
    default:
      return state;
  }
}
