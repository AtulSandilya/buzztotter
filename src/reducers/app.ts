const initialAppState = {
  isLoading: true,
}

export const app = (state = initialAppState, action) => {
  switch(action.type){
    case 'LOADING_COMPLETE':
      return Object.assign({}, state,
        {
          isLoading: false
        }
      );
    default:
      return state;
  }
}
