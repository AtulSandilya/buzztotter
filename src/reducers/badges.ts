import {sceneOrder, sceneKeys} from './view';

interface BadgesState {
  unseenReceivedBevegrams: number;
  upcomingBirthdays: number;
}

const initialState: BadgesState = {
  unseenReceivedBevegrams: 0,
  upcomingBirthdays: 0,
}

export const badges = (state: BadgesState = initialState, action) => {
  switch(action.type){
    case 'ADD_RECEIVED_BEVEGRAM_TO_BADGE':
      return Object.assign({}, state, {
        unseenReceivedBevegrams: state.unseenReceivedBevegrams + action.payload.quantity,
      })
    case 'RESET_BADGE':
      if(action.payload){
        const newScenePosition = action.payload.newScenePosition;
        switch(newScenePosition) {
          case sceneOrder[sceneKeys.bevegrams]:
            return Object.assign({}, state, {
              unseenReceivedBevegrams: 0,
            })
          default:
            return state
        }
      }
      return state;
    default:
      return state;
  }
}
