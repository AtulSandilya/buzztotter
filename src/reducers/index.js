import { combineReducers } from 'redux';

import { settings } from './settings';
import { view } from './view';

export default combineReducers({
  settings,
  view,
})
