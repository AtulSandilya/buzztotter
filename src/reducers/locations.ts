import {Location} from "../db/tables";

export const defaultLocationsState = []

export const locations = (state = defaultLocationsState, action) => {
  switch (action.type) {
    case "REFRESH_LOCATIONS":
      return action.payload.locations;
    default:
      return state;
  }
};
