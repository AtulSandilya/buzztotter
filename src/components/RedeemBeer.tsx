import * as React from "react";
import { Component } from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View
} from 'react-native';

import {Location} from '../db/tables';
import {LocationsMatch} from '../CommonUtilities';

import RouteWithNavBarWrapper from './RouteWithNavBarWrapper';
import TitleText from './TitleText';
import BevButton from './BevButton';

import {globalColors, globalStyles} from './GlobalStyles';

import {DeviceLocation} from '../reducers/redeemView';

export interface RedeemBeerProps {
  id?: string;
  name?: string;
  quantity?: number;
  currentLocation?: DeviceLocation;
  currentLocationBusinessName?: string;
  currentLocationLastModified?: string;
  getLocationFailed?: boolean;
  redeemConfirmed?: boolean;
  locations?: [Location];
  onRedeemClicked?(string): void;
  closeRedeem?(): void;
  updateLocation?(string): void;
}

interface RedeemBeerState {
  numDrinks?: number;
}

export default class RedeemBeer extends Component<RedeemBeerProps, RedeemBeerState> {
  constructor(props){
    super(props);
    this.state = {
      numDrinks: 1,
    };
  }

  componentDidMount() {
    this.updateLocation();
  }

  purchaseDrink() {
    if(this.props.getLocationFailed || this.props.currentLocationBusinessName === undefined){
      alert("You are not at an establishment that accepts bevegrams. Please see the map for establishments that accept bevegrams.");
      return;
    }

    // TODO: implement this end to end
    // this.props.onRedeemClicked(this.props.id);
    alert("Not implemented");
  }

  renderPurchaseConfirmed(){
    if(this.props.redeemConfirmed){
      return (
        <View>
          <View style={{flex: 1, alignItems: 'center', paddingTop: 20}}>
            <Text style={{color: globalColors.bevPrimary, fontSize: 30}}>1 Beer Redeemed!</Text>
          </View>
          <View style={{alignItems: 'flex-end', paddingTop: 10}}>
            <BevButton
              onPress={this.props.closeRedeem}
              text={"Close"}
              shortText={"Close"}
              label="Close Redeem Button"
              buttonFontSize={20}
            />
          </View>
        </View>
      )
    }
  }

  updateLocation() {
    try {
      navigator.geolocation.getCurrentPosition((position) => {
        const deviceLocation: DeviceLocation = {
          latitude: position.coords.latitude,
          longitude: position.coords.longitude,
        }

        const matchingLocations = this.props.locations.filter((business) => {
          const businessLocation = {
            latitude: business.latitude,
            longitude: business.longitude,
          }

          return LocationsMatch(deviceLocation, businessLocation, business.name);
        })

        let businessName: string;

        if(matchingLocations.length === 1){
          businessName = matchingLocations[0].name + "\n" + matchingLocations[0].address.split(",")[0] + "\n" + matchingLocations[0].address.split(", ").slice(1).join(", ")
        } else {
          businessName = undefined;
        }

        this.props.updateLocation(Object.assign({}, {
          currentLocation: deviceLocation,
          getLocationFailed: businessName === undefined,
          currentLocationBusinessName: businessName,
        }))
      }, undefined, {enableHighAccuracy: false, timeout: 20000, maximumAge: 1});
    } catch(e) {
      this.props.updateLocation(Object.assign({}, {
        location: undefined,
        getLocationFailed: true,
        currentLocationBusinessName: "",
      }))
    }
  }

  render() {
    return(
      <RouteWithNavBarWrapper>
        <View style={globalStyles.bevContainer}>
          <View style={globalStyles.bevLine}>
            <View style={globalStyles.bevLineLeft}>
              <Text style={globalStyles.bevLineTextTitle}>From:</Text>
            </View>
            <View style={globalStyles.bevLineRight}>
              <Text style={globalStyles.bevLineText}>{this.props.name}</Text>
            </View>
          </View>
          <View style={globalStyles.bevLine}>
            <View style={globalStyles.bevLineLeft}>
              <Text style={globalStyles.bevLineTextTitle}>Quantity:</Text>
            </View>
            <View style={globalStyles.bevLineRight}>
              <Text style={globalStyles.bevLineText}>{this.props.quantity}</Text>
            </View>
          </View>
          <View style={globalStyles.bevLine}>
            <View style={globalStyles.bevLineLeft}>
              <Text style={globalStyles.bevLineTextTitle}>Your Location:</Text>
            </View>
            <View style={globalStyles.bevLineRight}>
              <TouchableHighlight
                onPress={() => this.updateLocation()}
                underlayColor={'rgba(255, 255, 255, 0.1)'}
              >
                <View>
                  {this.props.currentLocationBusinessName ?
                    <Text
                      style={globalStyles.bevLineText}
                      numberOfLines={3}
                    >
                      {this.props.currentLocationBusinessName}
                    </Text>
                  :
                    <Text style={globalStyles.bevLineText} numberOfLines={2}>
                    {this.props.getLocationFailed ?
                      "Unable to determine your location"
                    :
                      "Retrieving Location..."
                    }
                    </Text>
                  }
                  <Text style={[globalStyles.bevTipText, {paddingTop: 5}]}>TAP TO REFRESH</Text>
                </View>
              </TouchableHighlight>
            </View>
          </View>
          <View style={{paddingBottom: 10}}>
            <View>
              <Text style={{color: 'red'}}>* Show this to your bartender or server:</Text>
              <Text>1. They enter the vendor id and press "Redeem Bevegram".</Text>
              <Text>2. They get you a nice cold beverage.</Text>
              <Text>3. You enjoy a nice cold beverage.</Text>
            </View>
          </View>
          <View style={globalStyles.bevLine}>
            <View style={globalStyles.bevLineLeft}>
              <Text style={globalStyles.bevLineTextTitle}>Vendor PIN:</Text>
            </View>
            <View style={[globalStyles.bevLineRight, {flex: 1, maxWidth: 125}]}>
              <TextInput
                placeholder={"1234"}
                placeholderTextColor={"#bbbbbb"}
                style={{flex: 1, textAlign: 'right', paddingRight: 10}}
                keyboardType={'numeric'}
              />
            </View>
          </View>
          <View style={{flexDirection: 'row', paddingTop: 20}}>
            <View style={{flex: -1, alignItems: 'flex-start'}}>
              <BevButton
                onPress={this.props.closeRedeem}
                text={"Close"}
                shortText={""}
                fontAwesomeLeftIcon="ban"
                label="Close Redeem Button"
                buttonFontSize={18}
              />
            </View>
            <View style={{flex: -1, alignItems: 'flex-end'}}>
              <BevButton
                onPress={this.purchaseDrink.bind(this)}
                text={`Redeem ${this.props.quantity} Bevegram${this.props.quantity === 1 ? "" : "s"}`}
                shortText={"Redeem"}
                label={"Redeem Bevegram Button"}
                buttonFontSize={18}
              />
            </View>
          </View>
          {this.renderPurchaseConfirmed()}
        </View>
      </RouteWithNavBarWrapper>
    );
  }
}
