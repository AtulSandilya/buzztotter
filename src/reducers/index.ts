import { combineReducers } from "redux";

import { reducer as network } from "react-native-offline";

import { addCreditCard } from "./addCreditCard";
import { app } from "./app";
import { badges } from "./badges";
import { banner } from "./banner";
import { bevegrams } from "./bevegrams";
import { bevegramsTab } from "./bevegramsTab";
import { contacts } from "./contacts";
import { contactsView } from "./contactsView";
import { historyView } from "./historyView";
import { locations } from "./locations";
import { locationsView } from "./locationsView";
import { login } from "./login";
import { loginView } from "./loginView";
import { message } from "./message";
import { modals } from "./modals";
import { purchase } from "./purchase";
import { purchasedBevegrams } from "./purchasedBevegrams";
import { receivedBevegrams } from "./receivedBevegrams";
import { redeem } from "./redeem";
import { redeemedBevegrams } from "./redeemedBevegrams";
import { redeemInProgressView } from "./redeemInProgressView";
import { redeemPickerView } from "./redeemPickerView";
import { redeemVendorIdView } from "./redeemVendorIdView";
import { routes } from "./routes";
import { sentBevegrams } from "./sentBevegrams";
import { settings } from "./settings";
import { user } from "./user";
import { view } from "./view";

const appReducer = combineReducers({
  addCreditCard,
  app,
  badges,
  banner,
  bevegrams,
  bevegramsTab,
  contacts,
  contactsView,
  historyView,
  locations,
  locationsView,
  login,
  loginView,
  message,
  modals,
  network,
  purchase,
  purchasedBevegrams,
  receivedBevegrams,
  redeem,
  redeemInProgressView,
  redeemPickerView,
  redeemVendorIdView,
  redeemedBevegrams,
  routes,
  sentBevegrams,
  settings,
  user,
  view,
});

const rootReducer = (state, action) => {
  if (action.type === "RESET_STATE") {
    state = undefined;
  }

  return appReducer(state, action);
};

export default rootReducer;
