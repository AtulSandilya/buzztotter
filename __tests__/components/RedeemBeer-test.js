import React from 'react';
import { Text, View } from 'react-native';

import { shallow } from 'enzyme';

import { defaultLocationsState } from '../../build/reducers/locations';

import RedeemBeer from '../../build/components/RedeemBeer';

describe('RedeemBeer component', () => {
  it('renders successfully', () => {
    const wrapper = shallow(
      <RedeemBeer
        id="1234"
        name="Testy"
        locations={defaultLocationsState}
      />
    )

    const numButtons = wrapper.find(View).find('BevButton').length;
    // Redeem Beer should have 2 buttons
    expect(numButtons).toEqual(2);
  })
})

