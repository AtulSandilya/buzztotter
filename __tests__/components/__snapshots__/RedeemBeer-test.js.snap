// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedeemBeer component renders correctly 1`] = `
<RCTScrollView
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={30}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="always"
  onScroll={[Function]}
  refreshControl={null}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "flex": 1,
      "height": NaN,
      "left": 0,
      "top": 85,
      "width": 750,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingBottom": 7.333333333333333,
          "paddingHorizontal": 13.472193585307478,
          "paddingTop": 11,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomWidth": 0.5,
            "borderColor": "#dddddd",
            "flex": -1,
            "flexDirection": "row",
            "marginBottom": 7.333333333333333,
            "paddingBottom": 7.333333333333333,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": -1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            fontWeight="bold"
            size="largeNormal"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#333333",
                "fontFamily": undefined,
                "fontSize": 18.371173070873834,
                "fontWeight": "bold",
                "letterSpacing": undefined,
                "lineHeight": 25.312499999999996,
                "marginLeft": 7.333333333333333,
                "textAlignVertical": "center",
              }
            }
            textStyle={
              Object {
                "marginLeft": 7.333333333333333,
              }
            }
          >
            From:
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            size="largeNormal"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#333333",
                "fontFamily": undefined,
                "fontSize": 18.371173070873834,
                "fontWeight": "normal",
                "letterSpacing": undefined,
                "lineHeight": 25.312499999999996,
                "textAlignVertical": "center",
              }
            }
            textStyle={undefined}
          >
            Testy
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 0.5,
            "borderColor": "#dddddd",
            "flex": -1,
            "flexDirection": "row",
            "marginBottom": 7.333333333333333,
            "paddingBottom": 7.333333333333333,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": -1,
                "justifyContent": "center",
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            fontWeight="bold"
            size="largeNormal"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#333333",
                "fontFamily": undefined,
                "fontSize": 18.371173070873834,
                "fontWeight": "bold",
                "letterSpacing": undefined,
                "lineHeight": 25.312499999999996,
                "marginLeft": 7.333333333333333,
                "textAlignVertical": "center",
              }
            }
            textStyle={
              Object {
                "marginLeft": 7.333333333333333,
              }
            }
          >
            Quantity:
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            size="largeNormal"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#333333",
                "fontFamily": undefined,
                "fontSize": 18.371173070873834,
                "fontWeight": "normal",
                "letterSpacing": undefined,
                "lineHeight": 25.312499999999996,
                "textAlignVertical": "center",
              }
            }
            textStyle={undefined}
          >
            1
          </Text>
          <View />
        </View>
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "#dddddd",
                "flex": -1,
                "flexDirection": "row",
                "marginBottom": 7.333333333333333,
                "paddingBottom": 7.333333333333333,
              },
              Object {
                "borderBottomWidth": 0,
                "marginBottom": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": -1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              fontWeight="bold"
              size="largeNormal"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333333",
                  "fontFamily": undefined,
                  "fontSize": 18.371173070873834,
                  "fontWeight": "bold",
                  "letterSpacing": undefined,
                  "lineHeight": 25.312499999999996,
                  "marginLeft": 7.333333333333333,
                  "textAlignVertical": "center",
                }
              }
              textStyle={
                Object {
                  "marginLeft": 7.333333333333333,
                }
              }
            >
              Select Your Location:
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flex": -1,
                        "flexDirection": "row",
                        "margin": 0,
                        "padding": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "paddingRight": 7.333333333333333,
                          "width": undefined,
                        },
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Object {
                            "color": "#8D8D9D",
                            "fontSize": 15,
                            "lineHeight": 15,
                          },
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <Text
                    accessible={true}
                    allCaps={true}
                    allowFontScaling={true}
                    color="#51515E"
                    disabled={false}
                    ellipsizeMode="tail"
                    fontWeight="bold"
                    isCondensed={true}
                    size="small"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#51515E",
                        "fontFamily": "Roboto",
                        "fontSize": 10,
                        "fontWeight": "bold",
                        "letterSpacing": 0.5,
                        "lineHeight": 10,
                        "margin": 0,
                        "paddingBottom": 0,
                        "paddingTop": 2,
                        "textAlignVertical": "center",
                      }
                    }
                    textStyle={
                      Object {
                        "margin": 0,
                        "paddingBottom": 0,
                        "paddingTop": 2,
                      }
                    }
                  >
                    REFRESH
                  </Text>
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#555555",
                  "fontSize": 30,
                  "fontWeight": "300",
                },
                Object {
                  "paddingVertical": 11,
                },
              ]
            }
          >
            Unable to determine your location!
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "padding": 7.333333333333333,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "flex": -1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": -1,
                      "flexDirection": "row",
                      "margin": 0,
                      "padding": 0,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "paddingRight": 7.333333333333333,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#8D8D9D",
                          "fontSize": 15,
                          "lineHeight": 15,
                        },
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allCaps={true}
                  allowFontScaling={true}
                  color="#51515E"
                  disabled={false}
                  ellipsizeMode="tail"
                  fontWeight="bold"
                  isCondensed={true}
                  size="small"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#51515E",
                      "fontFamily": "Roboto",
                      "fontSize": 10,
                      "fontWeight": "bold",
                      "letterSpacing": 0.5,
                      "lineHeight": 10,
                      "margin": 0,
                      "paddingBottom": 0,
                      "paddingTop": 2,
                      "textAlignVertical": "center",
                    }
                  }
                  textStyle={
                    Object {
                      "margin": 0,
                      "paddingBottom": 0,
                      "paddingTop": 2,
                    }
                  }
                >
                  REFRESH
                </Text>
                <View />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
